workers/common/src/main/kotlin/common/env/NpmRcGenerator.kt

```diff
@@ -51,21 +56,11 @@ class NpmRcGenerator : EnvironmentConfigGenerator<NpmDefinition> {
             Base64.encode(toByteArray())
 
         /**
-         * Generate a configuration setting for te given [key] if an environment [variable] exists either with the
-         * provided name of in lower case.
-         */
-        private fun settingFromEnv(key: String, variable: String): String? {
-            val value = System.getenv(variable) ?: System.getenv(variable.lowercase())
-
-            return value?.let { "$key=$it" }
-        }
-
-        /**
-         * Generate proxy settings to be included in the generated configuration file based on existing environment
-         * variables.
+         * Generate a part of the proxy configuration based on the given [key] and [value]. Do not output anything if
+         * the value is undefined.
          */
-        private fun generateProxySettings(): List<String> = proxySettings.mapNotNull { (key, variable) ->
-            settingFromEnv(key, variable)
+        private fun PrintWriter.printProxySetting(key: String, value: String?) {
+            value?.let { println("$key=$it") }
```

Comment:
What is the added value of this intermediate function ? Can't it be inlined ?

Reply:
It reduces some redundancy; without the function, the *null* check would have to be done for each property of the proxy configuration.